async-openai/src/config.rs

```diff
@@ -31,17 +33,28 @@ pub trait Config: Clone {
 pub struct OpenAIConfig {
     api_base: String,
     api_key: Secret<String>,
+    #[serde(deserialize_with = "deserialize_header_map")]
+    headers: HashMap<String, String>,
```

Comment:
Please use native `HeaderMap` instead of custom conversion with custom deserializer.